lib/hooks/useFormDelayContext.tsx

```diff
@@ -0,0 +1,119 @@
+import { logMessage } from "@lib/logger";
+import { FormElement, FormProperties } from "@lib/types";
+import { createContext, useContext, useState } from "react";
+
+interface FormDelay {
+  startTime: number;
+  requiredQuestions: number;
+}
+
+const timerDefault: FormDelay = {
+  startTime: 0,
+  requiredQuestions: 0,
+};
+
+const FormDelayContext = createContext<{
+  formDelay: FormDelay;
+  setFormDelay: React.Dispatch<React.SetStateAction<FormDelay>>;
+}>({
+  formDelay: timerDefault,
+  setFormDelay: () => {},
+});
+
+export const FormDelayProvider = ({ children }: { children: React.ReactNode }) => {
+  const [formDelay, setFormDelay] = useState(timerDefault);
+  return (
+    <FormDelayContext.Provider value={{ formDelay, setFormDelay }}>
+      {children}
+    </FormDelayContext.Provider>
+  );
+};
+
+// Adds in a little math to make it more unpredictable and calculates the final form delay
+const calculateSubmitDelay = (delayFromFormData: number) => {
+  const secondsBaseDelay = 2; // To help test group forms, make this big e.g. 20000
+  const secondsPerFormElement = 2;
+  return secondsBaseDelay + delayFromFormData * secondsPerFormElement;
+};
+
+export const calculateDelayWithGroups = (
+  startTime: number,
+  endTime: number,
+  requiredQuestions: number
+) => {
+  if (isNaN(startTime) || isNaN(endTime) || isNaN(requiredQuestions)) {
+    return -1;
+  }
+  const elapsedTime = Math.floor((endTime - startTime) / 1000);
+  const delayFromFormData = requiredQuestions - elapsedTime;
+  return calculateSubmitDelay(delayFromFormData);
+};
+
+export const calculateDelayWithoutGroups = (formElements: FormElement[]) => {
+  if (!Array.isArray(formElements)) {
+    return -1;
+  }
+  const delayFromFormData = formElements.filter(
+    (element) => element.properties.validation?.required === true
+  ).length;
+  return calculateSubmitDelay(delayFromFormData);
+};
+
+// Turn on for local testing
+const debug = false;
```

Comment:
Not sure why we would need such boolean. From what I am seeing it is only used in one place to trigger a log.

Is not there an environment variable that we could use to detect whether we are in local or not? I thought there was one but I would have to search for it.

Reply:
I found `process.env.NODE_ENV === "development"` in the `logger.ts` file